src/main/java/com/minecolonies/coremod/client/gui/WindowResearchTree.java

```diff
@@ -349,11 +349,12 @@ private void drawTreeBackground(final ZoomDragView view, final int maxHeight)
             final Text timeLabel = new Text();
             timeLabel.setSize(TIME_WIDTH, TIME_HEIGHT);
             timeLabel.setPosition((i - 1) * (GRADIENT_WIDTH + X_SPACING) + GRADIENT_WIDTH / 2 - TIME_WIDTH / 4, TIMELABEL_Y_POSITION);
+            final double hoursTime = ((Math.pow(2, i - 1) * IGlobalResearchTree.getInstance().getBranchData(branch).getBaseTime()) * BASE_RESEARCH_TIME) / (BASE_RESEARCH_TIME * 2);
```

Comment:
I feel like we should have a utility method for this somewhere

Reply:
`(Math.pow(2, i - 1) * IGlobalResearchTree.getInstance().getBranchData(branch).getBaseTime()) * BASE_RESEARCH_TIME)` could probably get wrapped into one, either in `BranchData.getBaseTime` or elsewhere.  It gives a weird unit, though (effectively 1/288 of an hour).  Not sure if there's a better place for it.

Reply:
getBranchData(branch).getHoursTime(i), maybe?

Reply:
I can add that as well, but this is the only spot that actually checks for hours directly.  The in-progress research estimates all do their math through immediate comparisons to current progress and then convert it into hours, because doing the division first seems like it'd be asking for IEEE745 weirdness.

Reply:
Hmm, true, let's keep this here then for now. Just keeping in mind that when we eventually need this anywhere else then we make a method for it.

Reply:
If it's likely to come up, or even be useful to others interacting through the API, it's pretty trivial to put in place.  And it definitely does help to have all of these math calls moved into one spot, especially since we may want to have alternative calculations some time in the future (if only for external data packs).